lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -891,39 +891,49 @@ namespace Js
             if (!threadContext->RecordImplicitException())
                 return FALSE;
-            JavascriptError::ThrowTypeError(GetScriptContext(), JSERR_ErrorOnRevokedProxy, _u("enumerate"));
-        }
-
-        //4. Let trap be the result of GetMethod(handler, "enumerate").
-        //5. ReturnIfAbrupt(trap).
-        //6. If trap is undefined, then
-        //a.Return the result of calling the[[Enumerate]] internal method of target.
-        //7. Let trapResult be the result of calling the[[Call]] internal method of trap with handler as the this value and a new List containing target.
-        //8. ReturnIfAbrupt(trapResult).
-        //9. If Type(trapResult) is not Object, then throw a TypeError exception.
-        //10. Return trapResult.
-        JavascriptFunction* getEnumeratorMethod = GetMethodHelper(PropertyIds::enumerate, scriptContext);
-        Assert(!GetScriptContext()->IsHeapEnumInProgress());
-        if (nullptr == getEnumeratorMethod)
-        {
-            return target->GetEnumerator(enumNonEnumerable, enumerator, requestContext, preferSnapshotSemantics, enumSymbols);
+            JavascriptError::ThrowTypeError(GetScriptContext(), JSERR_ErrorOnRevokedProxy, _u("ownKeys"));
+        }    
+        
+        Var enmeratorObj;
+        Var propertyName = nullptr;
+        PropertyId propertyId;
+        int index = 0;
+        JsUtil::BaseDictionary<const char16*, Var, Recycler> dict(scriptContext->GetRecycler());
+        JavascriptArray* arrResult = scriptContext->GetLibrary()->CreateArray();
+
+        // 13.7.5.15EnumerateObjectProperties(O)#
+        //  for (let key of Reflect.ownKeys(obj)) {    
+        Var trapResult = JavascriptOperators::GetOwnPropertyNames(this, scriptContext);
+        Assert(JavascriptArray::Is(trapResult));
+        ((JavascriptArray*)trapResult)->GetEnumerator(false, &enmeratorObj, scriptContext);
+        JavascriptEnumerator* pEnumerator = JavascriptEnumerator::FromVar(enmeratorObj);
+        while ((propertyName = pEnumerator->GetCurrentAndMoveNext(propertyId)) != NULL)
+        {
+            PropertyId  propId = JavascriptOperators::GetPropertyId(propertyName, scriptContext);
+            Var prop = JavascriptOperators::GetProperty(RecyclableObject::FromVar(trapResult), propId, scriptContext);
+            // if (typeof key === "string") {
+            if (JavascriptString::Is(prop))
+            {
+                Js::PropertyDescriptor desc;
+                JavascriptString* str = JavascriptString::FromVar(prop);
+                // let desc = Reflect.getOwnPropertyDescriptor(obj, key);
+                BOOL ret = JavascriptOperators::GetOwnPropertyDescriptor(this, str, scriptContext, &desc);
+                // if (desc && !visited.has(key)) {
+                if (ret && !dict.ContainsKey(str->GetSz()))
+                {
+                    dict.Add(str->GetSz(), prop);
+                    //if (desc.enumerable) yield key;
```

Comment:
for this one it's using pseudo code instead of items to explain the spec of this section.
And there's section number above. Added the link(including the clause id) to spec as well.